citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/CitationManager.java

```diff
@@ -86,6 +87,8 @@ public class CitationManager implements ICitationManager {
 
     private Map<String, BiFunction<ICitation, ICitation, Integer>> sortFunctions;
 
+    Future<String> future;
```

Comment:
this will result in only one future stored. If two people cancel two different jobs, you'll only have one future (for the job canceled last).